### PR TITLE
[extension-selenium] Do not wait for elements in assert element number verifications

### PR DESCRIPTION
**Pull request by uarlouski**
…r verifications

### CODE REVIEW

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java [hunk 1]
```diff
@@ -363,6 +363,7 @@ public boolean assertIfElementDoesNotExist(String businessDescription, String sy
     {
         return runValidatingSearchContext(searchContext, () ->
         {
+            locator.getSearchParameters().setWaitForElement(false);
```
[the remark this answers is not in the document]
Comment (uarlouski) [reply vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:366]: Hmm I assumed that all steps that check element number on screen should not wait for them, for this purpose we have wait steps

[the remark this answers is not in the document]
Comment (ikalinin1) [reply vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:366]: I propose to pass this as an option via locator, for example : id(someId):a:nw
or event id(someId):a:wf(PT1S)

[the remark this answers is not in the document]
Comment (uarlouski) [reply vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:366]: Please file a ticket for this improvement :)

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java [hunk 2]
```diff
@@ -363,6 +364,14 @@ public boolean assertIfElementDoesNotExist(String businessDescription, String sy
     {
         return runValidatingSearchContext(searchContext, () ->
         {
+            if (number == 0)
+            {
+                isTrue(ComparisonRule.LESS_THAN != comparisonRule, "Number of elements can not be negative");
+                if (ComparisonRule.EQUAL_TO == comparisonRule)
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:370]: we need to add `LESS_THAN_OR_EQUAL_TO` to this condition as well

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java [hunk 3]
```diff
@@ -363,6 +364,14 @@ public boolean assertIfElementDoesNotExist(String businessDescription, String sy
     {
         return runValidatingSearchContext(searchContext, () ->
         {
+            if (number == 0)
+            {
+                isTrue(ComparisonRule.LESS_THAN != comparisonRule, "Number of elements can not be negative");
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/validation/BaseValidations.java:369]: "Invalid input rule: the number of elements can not be less than 0"

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Pull request merged**